### DISCUSSION TITLE
Monospace font setting for all editors

### DISCUSSION THREAD

**Discussion by c-rush** [standardnotes/forum#2685] [General]
**Is your feature request related to a problem? Please describe.**
Markdown syntax can be difficult to read without a monospaced font (especially for tables)

**Describe the solution you'd like**
Minimum viable product: monospace font toggle under “General” settings can be extended to apply to all editors, not just Plain Text.
Ideal: ability to set desired font globally, with separate options for editors and preview panes. 

**Describe alternatives you've considered**
Editing in a separate text editor, but this completely negates the convenience of toggling the preview, real-time sync, etc.

Thank you for considering!

**Comment by scheidm** [comment]
Please fix this! There's even a monospace font option already implemented, but it doesn't work on Markdown! This is ludicrous for a product focused on privacy: Half of your userbase is likely employed in tech in some means!

**Comment by effieeee** [comment]
Hi, unfortunately, the 3rd party nature of the other editors does make it rather difficult to implement global customization features like these 😞 There are certain limits to what we're able to change on each editor as well, so we'll have to check on that, but of course, I'll keep this open for now!

**Comment by Sjeiti** [comment]
Btw, check this [comment](https://github.com/standardnotes/forum/issues/1798#issuecomment-1262441524) for a (temporary) plugin alternative.

How is this difficult to implement when there is a plugin that does exactly that?

**Comment by myreli** [comment]
> How is this difficult to implement when there is a plugin that does exactly that?

Hi, @Sjeiti. Just expanding on @effieeee's response,

SN provides the `--sn-stylekit-editor-font-family` variable, which editors can use to set the font according to user preferences. When you toggle "Monospace Font" on preferences, SN sets that variable from `--sn-stylekit-sans-serif-font` to `--sn-stylekit-monospace-font`. However, if an editor uses that variable or not is, unfortunately, out of reach of SN. 

While the theme can [override (`!important`)](https://github.com/DanielNetoP/markdown-monospace/blob/main/main.css#L4) the editors' stylesheet through CodeMirror classes, SN doing that could unintentionally break editors or other themes. While unfortunate, the only way to fix this is to change each editor to follow the correct variables.
